backend/azureblob/azureblob.go

```diff
@@ -1088,7 +1088,7 @@ func (f *Fs) list(ctx context.Context, containerName, directory, prefix string,
 			isDirectory := isDirectoryMarker(*file.Properties.ContentLength, file.Metadata, remote)
 			if isDirectory {
 				// Don't insert the root directory
-				if remote == directory {
+				if remote == f.opt.Enc.ToStandardPath(directory) {
```

Comment:
`directory` should already be in Standard encoding here since it ultimately came from the user rather than the backend, so I'm not sure what is going on here.

Same for the other backends.

Reply:
I'm not sure about that. For example here:

https://github.com/rclone/rclone/blob/e9e9feb21eeb200782180e8532f2a0a95f745740/backend/s3/s3.go#L4178

`directory` is returned from `f.split`, which does `FromStandardPath` on it:

https://github.com/rclone/rclone/blob/e9e9feb21eeb200782180e8532f2a0a95f745740/backend/s3/s3.go#L2823-L2826

Here's a test to reproduce the issue:

```
go test -timeout 3h -run '^TestBisyncRemoteRemote/extended_filenames$' github.com/rclone/rclone/cmd/bisync -remote TestS3,directory_markers: -v 
```

Reply:
Yes I've checked the 3 backends in question and you are 100% right :-)

I'll cherry pick this commit now.